pco_python/src/standalone.rs

```diff
@@ -16,23 +17,25 @@ fn decompress_chunks<'py, T: NumberLike + Element>(
   mut src: &[u8],
   file_decompressor: FileDecompressor,
 ) -> PyResult<&'py PyArray1<T>> {
-  let n_hint = file_decompressor.n_hint();
-  let mut res: Vec<T> = Vec::with_capacity(n_hint);
-  while let MaybeChunkDecompressor::Some(mut chunk_decompressor) = file_decompressor
-    .chunk_decompressor::<T, &[u8]>(src)
-    .map_err(pco_err_to_py)?
-  {
-    let initial_len = res.len(); // probably always zero to start, since we just created res
-    let remaining = chunk_decompressor.n();
-    unsafe {
-      res.set_len(initial_len + remaining);
-    }
-    let progress = chunk_decompressor
-      .decompress(&mut res[initial_len..])
-      .map_err(pco_err_to_py)?;
-    assert!(progress.finished);
-    src = chunk_decompressor.into_src();
-  }
+  let res = py
+    .allow_threads(|| -> PcoResult<Vec<T>> {
```

Comment:
You probably don't need the type signature here? Or if you do for type inference reasons, better to place it on `res: ...` I think.